### ISSUE TITLE
API Pain-points

### ISSUE THREAD

**Issue by jakevdp** [altair-viz/altair#947]
This is a running tally, in no particular order:

- [ ] ``sort`` options in encodings. Awkward because ``alt.SortField``/``alt.EncodingSortField`` require an ``op`` parameter even if one isn't needed. Maybe a good candidate for a shorthand syntax?
- [ ] ``transform_bin``. Awkward because it doesn't follow the pattern of ``transform_calculate``, and the field is specified as a string
- [ ] ``transform_window()``. Awkward because it requires a list of dicts, and those dicts include the keyword ``as`` so the associated objects cannot be used.
- [ ] normalized histogram. So many transforms required... see https://groups.google.com/forum/#!topic/altair-viz/NMynW-PUqZk

**Comment by jeffbaumes** [comment]
I have not been using the Altair API too long but +1 to `chart.facet()`. It is tough to keep composition options (layer, facet, repeat) straight. For dealing with the chart later after a facet or other composition you have to know when to reach into the `spec` or `layer` object, where to put `data`, etc. I seem to get the "required field 'data' missing" frequently. There are also some other gotchas (can't facet a facet, facet ordering does not seem possible, config values cannot be inside the facet spec but need to be at the top level).

However, this is complexity inherent in Vega Lite that may be difficult/impossible to hide in Altair.

**Comment by jakevdp** [comment]
Yeah, this issue is not intended to be about pain points in the grammar, but pain points in how we map the grammar to Python.

**Comment by jakevdp** [comment]
> It would be nicer to have something more in line with the other configure options, for example:
.configure_title(color='red)

That should work in the current release.

**Comment by gwdekker** [comment]
Apologies, you are right it does - it seems it is just the documentation that is not up to date on that point which threw me off.
